spatial/r3/mat_safe.go

```diff
@@ -136,3 +136,25 @@ func arrayFrom(vals []float64) *array {
 	copy(a[:], vals)
 	return &a
 }
+
+// Mat returns a 3×3 rotation matrix corresponding to the receiver. It
+// may be used to perform rotations on a 3-vector or to apply the rotation
+// to a 3×n matrix of column vectors. If the receiver is not a unit
+// quaternion, the returned matrix will not be a pure rotation.
+func (r Rotation) Mat() *Mat {
+	re, im, jm, km := r.Real, r.Imag, r.Jmag, r.Kmag
```

Comment:
It's just naming but since we're touching this I'd prefer:
```
w, i, j, k := ...
ii := 2*i*i
jj := 2*j*j
...
```

Reply:
Done.